godot-core/src/log.rs

```diff
@@ -77,15 +77,31 @@ macro_rules! godot_print {
     };
 }
 
-pub use crate::{godot_error, godot_print, godot_script_error, godot_warn};
+#[macro_export]
+macro_rules! godot_print_rich {
+    ($fmt:literal $(, $args:expr)* $(,)?) => {
+        $crate::log::print(true, &[
+            $crate::builtin::Variant::from(
+                $crate::builtin::GString::from(
+                    format!($fmt $(, $args)*)
+                )
+            )
+        ])
+    };
+}
+
+pub use crate::{godot_error, godot_print, godot_print_rich, godot_script_error, godot_warn};
 
 use crate::builtin::{StringName, Variant};
 use crate::sys::{self, GodotFfi};
 
-/// Prints to the Godot console, used by the [`godot_print!`] macro.
-pub fn print(varargs: &[Variant]) {
+/// Prints to the Godot console, used by the [`godot_print!`] and  [`godot_print_rich!`] macros.
+pub fn print(use_bbcode: bool, varargs: &[Variant]) {
     unsafe {
-        let method_name = StringName::from("print");
+        let method_name = match use_bbcode {
+            false => StringName::from("print"),
+            true => StringName::from("print_rich"),
+        };
```

Comment:
This is a public function, adding this `bool` parameter breaks compatibility and makes it more cumbersome to use.

Why not add a new internal method that takes the `StringName` as a parameter?

Reply:
Alternatively, we could probably remove this method, since there is `utilities::print` and `utilities::print_rich`:
https://godot-rust.github.io/docs/gdext/master/godot/engine/utilities/index.html

Maybe these could be used directly internally (just passing 1 argument, the formatted string).

The current implementation also suffers from a different problem: wrong hash. This wouldn't be necessary if we reuse the utility functions.

Reply:
I'll try using the utilities methods. About the internal method print(), I thought it was added to be only used within the godot_print! (so maybe the doc should be hidden?). If so, adding the bool should not be a problem. This being said, I'm new to this so may be wrong.

Reply:
It was added before we had bindings for the utility methods, to provide a function (not just a macro) for printing.

The generated utility methods are still a bit strange, as their signature is:
```rs
pub fn print(arg1: Variant, varargs: &[Variant]);
```
but that's what the GDExtension provides. I think most people use the macros anyway, and if not, this should be an OK alternative.